cmd/mysterium_client/mysterium_client.go

```diff
@@ -18,40 +18,42 @@ func main() {
 		os.Exit(1)
 	}
 
-	cmdRun := client.NewCommand(options)
-	cmd.NewTerminator(cmdRun)
+	var cmdRun *client.Command
+	var cmdCli *cli.Command
+
+	stop := func() {
+		var killers []cmd.Killer
+		if cmdRun != nil {
+			killers = append(killers, cmdRun.Kill)
+		}
+		// TODO: add CLI killer once it's not blocking anymore
+		cmd.NewApplicationStopper(killers...)
+		os.Exit(0)
+	}
+
+	cmdRun = client.NewCommand(options, stop)
 
 	if err := cmdRun.Run(); err != nil {
 		fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
 
 	if options.CLI {
-		cmdCli := cli.NewCommand(
+		cmdCli = cli.NewCommand(
 			filepath.Join(options.DirectoryRuntime, ".cli_history"),
 			tequilapi_client.NewClient(options.TequilapiAddress, options.TequilapiPort),
-			newStopHandler(cmdRun),
+			stop,
 		)
-		cmd.NewTerminator(cmdCli)
 		if err := cmdCli.Run(); err != nil {
 			fmt.Fprintln(os.Stderr, err)
 			os.Exit(1)
 		}
 	}
 
+	cmd.NewTerminator(stop)
```

Comment:
this place could register 2nd sigterm handler together with CLI

Reply:
We can't invoke `Kill` of cli command, because it blocks.

Reply:
I mean 2 sig handlers

Reply:
Inspected that signals handled by us and by `Readline` do not overlap.